hal/src/electron/rtos/FreeRTOSv8.2.2/FreeRTOS/Source/portable/MemMang/heap_4_lock.c

```diff
@@ -324,8 +363,8 @@ void *pvReturn = NULL;
     {
         if( pvReturn == NULL )
         {
-            extern void vApplicationMallocFailedHook( void );
-            vApplicationMallocFailedHook();
+            extern void vApplicationMallocFailedHook( size_t );
+            vApplicationMallocFailedHook(xWantedSize);
```

Comment:
Shouldn't we report the originally requested size here (`xWantedSize - xHeapStructSize`) similarly to the Photon implementation?

Another problem here is that FreeRTOS invokes `vApplicationMallocFailedHook()` when the size passed to `pvPortMalloc ()` is 0. Given that our current `malloc()` is a trivial wrapper over `pvPortMalloc()` this would cause the `out_of_memory` event to be generated for the `void* ptr = malloc(0)` statement, which is perfectly valid in terms of the standard library.

Reply:
yes, in fact, I wanted to move that file to a common location and avoid the duplication. I will do that.  And I will add a 0 size check so that the system event is not fired. Thanks for picking that up - really good catch!